internal/dag/builder.go

```diff
@@ -515,15 +513,21 @@ func (b *builder) computeIngressRoutes() {
 		if tls := ir.Spec.VirtualHost.TLS; tls != nil {
 			// attach secrets to TLS enabled vhosts
 			m := splitSecret(tls.SecretName, ir.Namespace)
-			if sec := b.lookupSecret(m, validSecret); sec != nil && b.delegationPermitted(m, ir.Namespace) {
+			sec, secretInvalidOrNotFound := b.lookupSecret(m, validSecret)
+			if sec != nil && b.delegationPermitted(m, ir.Namespace) {
```

Comment:
This is the only place that the second boolean parameter is begin used. Would this workinstead

```go
sec := b.lookupSecret(m, validSecret)
secretInvalidOrNotFound := sec == nil
```

Reply:
Yup that works! For some reason, I was making that more complicated than it needed to be.